app/views/layouts/application.html.erb

```diff
@@ -0,0 +1,23 @@
+<!DOCTYPE html>
+<html>
+<head>
+  <title>InstagramChallenge</title>
+  <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track' => true %>
+  <%= javascript_include_tag 'application', 'data-turbolinks-track' => true %>
+  <%= csrf_meta_tags %>
+</head>
+<body>
+  <% if user_signed_in? %>
+     <%= link_to "Sign out", destroy_user_session_path, method: :delete %>
+   <% else %>
+     <%= link_to "Sign in", new_user_session_path %>
+     <%= link_to "Sign up", new_user_registration_path %>
+   <% end %>
```

Comment:
Maybe extract this to a partial called `_navigation.erb`. You can render this partial anywhere using:

```ruby
<%= render 'navigation' %>
```